docs/01-app/02-building-your-application/07-configuring/05-mdx.mdx

```diff
@@ -248,6 +248,72 @@ export default function Page() {
 
 Navigating to the `/mdx-page` route should display your rendered MDX page.
 
+<AppOnly>
+
+### Using dynamic imports
+
+You can also dynamically import MDX in your pages. This is useful for:
+
+- Programmatically rendering MDX.
+- Pairing it with dynamic route segments and [`generateStaticParams`](/docs/app/api-reference/functions/generate-static-params).
+
+Create a new dynamic page within the `/app` directory and add MDX files wherever you'd like:
```

Comment:
```suggestion
You can import dynamic MDX components instead of using filesystem routing for MDX files.

For example, you can have a dynamic route segment which loads MDX components from a separate directory:
```

Reply:
@leerob Oh, was the suggestion here not good for some reason? It originally came from this discussion with @delbaoliveira , including my comment about "Programmatically importing some content (eg. programmatic logic beyond a simple Dynamic Segment like `[slug]/page.mdx`)":

- https://github.com/vercel/next.js/pull/73466#discussion_r1867546629

Maybe the bullet above is too vague / obscure / jargony in your opinion? I can understand that "programmatically" can be jargony and abstract.

To be clear, the pattern that I was suggesting is something like this:

`app/appointments/[slug]/page.tsx`

```tsx
type Props = {
  params: Promise<{
    slug: string;
  }>;
};

export default function AppointmentPage(props: Props) {
  let MDXContent;

  // Simplified example
  if (isLecture()) {
    MDXContent = (await import(`@/content/lectures/${slug}/index.mdx`)).default;
  } else {
    MDXContent = (await import(`@/content/appointments/${slug}/index.mdx`)).default;
  }

  return <MDXContent />;
}
```

I guess another way of expressing this particular use case would be "conditional MDX imports" - although I guess that there may be other "programmatic" use cases.

Reply:
Programatically felt unnecessary to explain to the code. "dynamically" here should accomplish the same goal.